activerecord/lib/active_record/relation/predicate_builder/association_query_value.rb

```diff
@@ -31,9 +31,8 @@ def primary_key
         end
 
         def convert_to_id(value)
-          case value
-          when Base
-            value._read_attribute(primary_key)
+          if value.respond_to?(primary_key)
+            value.public_send(primary_key)
```

Comment:
```suggestion
          if value.respond_to?(:id)
            value.id
```

Reply:
@rafaelfranca thanks for the review! Is it important to undo support for custom primary keys added by https://github.com/rails/rails/commit/cd0ed12d1a9edc73be953d700748b5827df890c7? Is `#id` an alias for `#primary_key` in that case?

Reply:
`id` is an alias to `_read_attribute(primary_key)` http://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/PrimaryKey.html#method-i-id

Reply:
Thanks @rafaelfranca. I've pushed up https://github.com/rails/rails/commit/98ee2a40bf.